### PR TITLE
engine: update to v27.1.0

### PR DESCRIPTION
**Pull request by thaJeztah**
## Description

- Add release notes
- Update examples in installation section
- Align some formatting between distros in the install section

## Reviews

- [ ] Technical review
- [ ] Editorial review
- [ ] Product review

### CODE REVIEW

File: content/engine/install/raspberry-pi-os.md [hunk 1]
```diff
@@ -53,9 +53,9 @@ To install Docker Engine, you need one of the following OS versions:
 
 Before you can install Docker Engine, you need to uninstall any conflicting packages.
 
-Distro maintainers provide an unofficial distributions of Docker packages in
-APT. You must uninstall these packages before you can install the official
-version of Docker Engine.
+Distro maintainers provide unofficial distributions of Docker packages in
```
Comment (thaJeztah) [inline content/engine/install/raspberry-pi-os.md:56]: Why does CI invalidate something that was already there (and is used in the other variants of this page?

Screenshot 2024-07-23 at 17 28 12

Comment (thaJeztah) [reply content/engine/install/raspberry-pi-os.md]: Reverted this change for now, but we should look how we can tweak these 🤔

**Comment by thaJeztah** [comment]
Wait; why vale all of a sudden green, but still reporting things as "error". Well. I guess ✅

File: content/engine/release-notes/27.1.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- CLI: add and improve shell completions for various flags [moby/moby#5261](https://github.com/moby/moby/pull/5261)
+- CLI: add OOMScoreAdj to `docker service create` and `docker stack` [docker/cli#5274](https://github.com/docker/cli/pull/5274)
+- CLI: add support for `DOCKER_CUSTOM_HEADERS` environment variable (experimental) [docker/cli#5271](https://github.com/docker/cli/pull/5271)
+- CLI: containerd-integration: Fix `docker push` defaulting the `--platform` flag to a value of `DOCKER_DEFAULT_PLATFORM` environment variable on unsupported API versions [docker/cli#5248](https://github.com/docker/cli/pull/5248)
+- CLI: fix: context cancellation on `login` prompt [docker/cli#5260](https://github.com/docker/cli/pull/5260)
+- CLI: fix: wait for the container to exit before closing the stream when sending a termination request to the CLI while attached to a container [docker/cli#5250](https://github.com/docker/cli/pull/5250)
+
+### Deprecated
+
+- The `pkg/rootless/specconv` package is deprecated, an will be removed in the next release [moby/moby#48185](https://github.com/moby/moby/pull/48185)
```
Comment (craig-osterhout) [inline content/engine/release-notes/27.1.md:49]:
```suggestion
- The `pkg/rootless/specconv` package is deprecated, and will be removed in the next release [moby/moby#48185](https://github.com/moby/moby/pull/48185)
```

File: content/engine/release-notes/27.1.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- The `pkg/containerfs` package is deprecated, an will be removed in the next release [moby/moby#48185](https://github.com/moby/moby/pull/48185)
```
Comment (craig-osterhout) [inline content/engine/release-notes/27.1.md:50]:
```suggestion
- The `pkg/containerfs` package is deprecated, and will be removed in the next release [moby/moby#48185](https://github.com/moby/moby/pull/48185)
```

File: content/engine/release-notes/27.1.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+- The `pkg/directory` package is deprecated, an will be removed in the next release [moby/moby#48185](https://github.com/moby/moby/pull/48185)
```
Comment (craig-osterhout) [inline content/engine/release-notes/27.1.md:51]:
```suggestion
- The `pkg/directory` package is deprecated, and will be removed in the next release [moby/moby#48185](https://github.com/moby/moby/pull/48185)
```

File: content/engine/release-notes/27.1.md (hunk 2 above)
Comment (thaJeztah) [reply content/engine/release-notes/27.1.md]: oh; good catch 🙈 should be fixed; PTAL 🤗

**Review by craig-osterhout — APPROVED** [review]
LGTM

**Pull request merged**